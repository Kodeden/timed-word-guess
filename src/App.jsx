import { useState } from "react";
import { Game, Setup } from "./routes";
import Layout from "./layout";

function App() {
  const [gameSetup, setGameSetup] = useState({});

  const handleSubmit = (e) => {
    e.preventDefault();

    // Gather the form details, matching the values to the names of the inputs (HTML)
    const formData = new FormData(e.target);

    // Convert the formData to an object
    const data = Object.fromEntries(formData);

    // Set the gameDeets state to the data
    setGameSetup(data);
  };

  return (
    <Layout>
      <Setup handleSubmit={handleSubmit} />
    </Layout>
  );
}

export default App;
