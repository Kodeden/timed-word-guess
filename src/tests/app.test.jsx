import { fireEvent } from "@testing-library/dom";
import { render, screen } from "@testing-library/react";
import App from "../App";
import { setup } from "./utils";

it("renders the setup screen", () => {
  const setupRender = render(<App />);
  expect(setupRender).toMatchSnapshot();
});

it("calls the player a loser", async () => {
  const { user } = setup(<App />);

  const wordTextInput = screen.getByLabelText(/word/i);
  const maxGuessesRangeInput = screen.getByLabelText(/max guesses/i);
  const submitButton = screen.getByRole("button", { name: /go/i });

  await user.type(wordTextInput, "hello");

  // AFAIK, we can't do range sliders with 'userEvent' yet
  // https://github.com/testing-library/user-event/issues/871#issuecomment-1059317998
  fireEvent.change(maxGuessesRangeInput, { target: { value: 5 } });

  await user.click(submitButton);

  // Wait for the underscores to appear ('game' test ✅ already verifies this)
  await screen.findByText(/_/i);

  // Make 5 incorrect guesses
  await user.keyboard("asdfg");

  // Wait for the 'loser' message to appear
  await screen.findByText(/lost/i);
});

it("calls the player a winner", async () => {
  const { user } = setup(<App />);

  const wordTextInput = screen.getByLabelText(/word/i);
  // We're not doing any wrong guesses anyway
  const submitButton = screen.getByRole("button", { name: /go/i });

  await user.type(wordTextInput, "hello");
  await user.click(submitButton);

  // Wait for the underscores to appear ('game' test ✅ already verifies this)
  await screen.findByText(/_/i);

  await user.keyboard("hello");

  // Wait for the 'winner' message to appear
  await screen.findByText(/won/i);
});
