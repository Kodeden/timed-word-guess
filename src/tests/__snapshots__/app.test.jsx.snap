// Vitest Snapshot v1

exports[`renders the setup screen 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="my-4 text-center text-2xl md:text-5xl"
      >
        Hangman Game
      </h1>
      <main
        class="p-4"
      >
        <form
          class="flex flex-col items-center space-y-4"
        >
          <div
            class="flex w-full flex-col items-center gap-y-2 text-center"
          >
            <label
              class="mr-4 font-medium"
              for="word"
            >
              Enter the Word or Phrase to Guess
            </label>
            <input
              class="w-full max-w-fit rounded-sm border-2 border-none border-gray-300 p-2 text-gray-800"
              id="word"
              name="word"
              required=""
              type="text"
            />
          </div>
          <div
            class="flex flex-col gap-y-1 border-b pb-2"
          >
            <label
              for="maxGuesses"
            >
              Max Guesses
            </label>
            <input
              id="maxGuesses"
              max="10"
              min="1"
              name="maxGuesses"
              step="1"
              type="range"
              value="1"
            />
            <p>
              Value: 
              <output
                id="value"
              >
                1
              </output>
            </p>
          </div>
          <div
            class="flex flex-col gap-y-1 border-b pb-2"
          >
            <label
              for="maxTime"
            >
              Max Time (minutes)
            </label>
            <input
              id="maxTime"
              max="10"
              min="1"
              name="maxTime"
              step="1"
              type="range"
              value="1"
            />
            <p>
              Value: 
              <output
                id="value"
              >
                1
              </output>
            </p>
          </div>
          <button
            class="button"
            type="submit"
          >
            Go!
          </button>
        </form>
      </main>
    </div>
  </body>,
  "container": <div>
    <h1
      class="my-4 text-center text-2xl md:text-5xl"
    >
      Hangman Game
    </h1>
    <main
      class="p-4"
    >
      <form
        class="flex flex-col items-center space-y-4"
      >
        <div
          class="flex w-full flex-col items-center gap-y-2 text-center"
        >
          <label
            class="mr-4 font-medium"
            for="word"
          >
            Enter the Word or Phrase to Guess
          </label>
          <input
            class="w-full max-w-fit rounded-sm border-2 border-none border-gray-300 p-2 text-gray-800"
            id="word"
            name="word"
            required=""
            type="text"
          />
        </div>
        <div
          class="flex flex-col gap-y-1 border-b pb-2"
        >
          <label
            for="maxGuesses"
          >
            Max Guesses
          </label>
          <input
            id="maxGuesses"
            max="10"
            min="1"
            name="maxGuesses"
            step="1"
            type="range"
            value="1"
          />
          <p>
            Value: 
            <output
              id="value"
            >
              1
            </output>
          </p>
        </div>
        <div
          class="flex flex-col gap-y-1 border-b pb-2"
        >
          <label
            for="maxTime"
          >
            Max Time (minutes)
          </label>
          <input
            id="maxTime"
            max="10"
            min="1"
            name="maxTime"
            step="1"
            type="range"
            value="1"
          />
          <p>
            Value: 
            <output
              id="value"
            >
              1
            </output>
          </p>
        </div>
        <button
          class="button"
          type="submit"
        >
          Go!
        </button>
      </form>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
